File: pset6/readability/readability.py
from cs50 import get_string

# prompt for input
text = get_string("Text: ")

words = len(text.split())
letters = 0
sentences = 0

# calculate letters and sentences
for c in text:
    if c.isalpha():
        letters += 1
    if c == "." or c == "!" or c == "?":
        sentences += 1

# calculate index
L = 100 * letters/words     # average number of letters per 100 words
S = 100 * sentences/words   # average number of sentences per 100 words in the text

index = 0.0588 * L - 0.296 * S - 15.8

# output the index
if index < 1:
    print("Before Grade 1")
elif index >= 16:
    print("Grade 16+")
else:
    print(f"Grade {round(index)}")

File: pset2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>


int main(void)
{
    // set up variables
    float letters = 0;
    float spaces = 0;
    float sentences = 0;

    // prompt for text
    string text = get_string("Text: ");


    // count letters
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isalpha(text[i]))
        {
            letters++;
        }
    }

    // count words (spaces)
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] == ' ')
        {
            spaces++;
        }
    }
    int words = spaces + 1;

    // count sentences
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }
    }


    // plug values into formula
    float L = (letters / words) * 100;    // L is average number of letters per 100 words
    float S = (sentences / words) * 100;    // S is average number of sentences per 100 words


    float index = 0.0588 * L - 0.296 * S - 15.8;

    // final result
    if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %.0f\n", index);
    }
}